cluster/log-dump/log-dump.sh

```diff
@@ -312,8 +329,25 @@ function dump_nodes_with_logexporter() {
     return
   fi
 
-  # Give some time for the pods to finish uploading logs.
-  sleep "${logexport_sleep_seconds}"
+  # Periodically fetch list of already logexported nodes to verify
+  # if we aren't already done.
+  start="$(date +%s)"
+  while true; do
+    now="$(date +%s)"
+    if [[ $((now - start)) -gt ${logexport_sleep_seconds} ]]; then
+      echo "Waiting for all nodes to be logexported timed out."
+      break
+    fi
+    if nodes_succeeded=$(gsutil ls ${gcs_artifacts_dir}/logexported-nodes-registry); then
```

Comment:
Maybe move this logic of listing files, etc into the nonexported_nodes function? So that you just check 'NON_LOGEXPORTED NODES' here.

Reply:
That was conscious - I wanted this function to be simple without retries etc (I wanted to avoid returning multiple things from it). OTOH, below we are reusing it but we do listing differently (errors would be handled differently in those two cases).

Reply:
Sorry, I meant that the logic of computing `LOGEXPORTED_NODES` and `NON_LOGEXPORTED_NODES` be moved into the function. So basically moving the `gsutil list` into that function (so no retries involved). Does that sgty?